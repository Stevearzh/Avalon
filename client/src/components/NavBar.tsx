import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import { withStyles, WithStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Menu from '@material-ui/icons/Menu';
import * as React from 'react';

const styles = {
  'nav-bar': {
    flexGrow: 1,
  },
  'menu-button': {
    marginLeft: -12,
    marginRight: 20,
  },
};

export interface Props extends WithStyles<typeof styles> {}

class NavBar extends React.Component<Props> {
  public render() {
    const { classes } = this.props;

    return (
      <div className={classes['nav-bar']}>
        <AppBar position="static">
          <Toolbar>
            <IconButton className={classes['menu-button']} color="inherit" aria-label="Menu">
              <Menu />
            </IconButton>
            <Typography variant="h6" color="inherit">
              IRC Logs
            </Typography>
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

export default withStyles(styles)(NavBar);
