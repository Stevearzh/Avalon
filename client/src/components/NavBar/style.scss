.nav-bar {
  position: fixed;
  width: 100%;
  top: 0;
  z-index: 10;

  .detail-info {
    display: flex;    
    float: right;
    position: absolute;
    top: 1em;
    right: 2em;
  }

  .input-label {
    color: #fff;
    margin-right: 0.5em;
    line-height: 32px;
  }

  .select-bar {    
    width: 100px;
    color: #fff;
    margin-right: 1.5em;

    &:before {
      background-color: #fff;
    }

    &:hover:before {
      background-color: #fff !important;
    }

    svg {
      color: #fff;
    }
  }

  .date-picker {
    display: flex;

    .picker-label {
      line-height: 32px;
      margin-right: 0.5em;
    }

    .picker > div {
      color: #fff;

      &:before {
        background-color: #fff;
      }
    }
  }
}
