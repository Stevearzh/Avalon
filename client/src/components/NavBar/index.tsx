import * as React from 'react';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import IconButton from 'material-ui/IconButton';
import MenuIcon from 'material-ui-icons/Menu';
import Input, { InputLabel } from 'material-ui/Input';
import Select from 'material-ui/Select';
import { MenuItem } from 'material-ui/Menu';
import * as moment from 'moment';
import { DatePicker } from 'material-ui-pickers/src';

import './style.scss';

interface Props {
  selectedDate: moment.Moment;
  selectedChannel: string;
  onDateChange: Function;
  onChannelChange: Function;  
}

interface State {
  channels: string[];
  minDate: string;
  availableDate: string[];
}

class NavBar extends React.Component<Props, State> {
  state = {
    channels: [],
    minDate: '1900-1-1',
    availableDate: []
  };

  componentDidMount () {
    fetch('/api/channels')
      .then((res: Response) => res.json())
      .then(json => this.setState({ channels: json.data.list }))
      .then(() => this.props.onChannelChange(this.state.channels[0]))
      .catch((error: {}) => null);

    fetch('/api/times')
      .then((res: Response) => res.json())
      .then(json => this.setState({ availableDate: json.data.list }))
      .then(() => this.setState({ minDate: `${this.state.availableDate[0]}-1` }))
      .catch(error => null);
  }

  handleChange = (event: React.ChangeEvent<HTMLInputElement>): void =>
    this.props.onChannelChange(event.target.value)

  handleDateChange = (date: moment.Moment): void =>
    this.props.onDateChange(date)

  handleDateDismiss = (event: React.MouseEvent<HTMLElement>): void => undefined;

  render () {
    return (
      <div className="nav-bar">
        <AppBar position="static">
          <Toolbar>
            <IconButton className="menu-button" color="contrast" aria-label="Menu">
              <MenuIcon />
            </IconButton>
            <div className="detail-info">
              <InputLabel
                htmlFor="channel-simple"
                className="input-label"
              >
                Channel:
              </InputLabel>
              <Select
                value={this.props.selectedChannel}              
                input={<Input name="channel" id="channel-simple" />}
                onChange={this.handleChange}
                className="select-bar"                
              >
                {
                  this.state.channels.map((channel: string, i: number) => (
                    <MenuItem value={channel} key={i}>
                      <em>{channel}</em>
                    </MenuItem>
                  ))
                }
              </Select>            
              <div className="date-picker">
                <div className="picker-label">Date:</div>
                <DatePicker
                  className="picker"
                  value={this.props.selectedDate}
                  onChange={this.handleDateChange}
                  disableFuture={true}
                  minDate={this.state.minDate}       
                />
              </div>
            </div>
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

export default NavBar;