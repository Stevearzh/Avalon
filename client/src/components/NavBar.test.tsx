import { shallow } from 'enzyme';
import * as React from 'react';

import NavBar from './NavBar';

it('matches snapshot', () => {
  expect(shallow(<NavBar theme="dark" onThemeChange={jest.fn()} />)).toMatchSnapshot();
});
