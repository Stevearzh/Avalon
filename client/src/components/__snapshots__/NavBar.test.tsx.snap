// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<NavBar
  classes={
    Object {
      "menu-button": "NavBar-menu-button-2",
      "nav-bar": "NavBar-nav-bar-1",
    }
  }
/>
`;
